### ISSUE TITLE
[BUG]: Ace Combat 04 - Glitchy ground textures near edges of screen

### ISSUE THREAD

**Issue by PositronCannon** [PCSX2/pcsx2#10976]
### Describe the Bug

I have a feeling this problem is probably known since it's been around for as long as I can remember, but I couldn't find an issue related to it here, so here goes.

As best demonstrated in [this video](https://www.youtube.com/watch?v=igLxJGXrI3k), the ground textures near the edges of the screen become garbled or missing, mostly in missions with mountainous terrain like mission 3 or (yet again) 11. This is most visible when using the game's built-in 16:9 mode as it expands the FoV horizontally, but it also affects the 4:3 mode (especially in first person view as it again has a wider FoV than third person). Note that in missions with mostly flat terrain this problem is much less visible or even absent entirely.

This happens on any HW renderer, and it does not reproduce in software mode.

As mentioned, this issue has been here for as long as I've been using PCSX2 (so at least the 1.5 days), but I feel like it's gotten worse in the past few months, though I have no hard proof of this.

You can find a GS dump attached below.

As far as I can tell, this is the last graphical issue remaining in AC04 HW that is not related to upscaling and is particularly noticeable (the other one I know is the sun shining through objects, but that's whatever in a game like this), so it'd be great if it can be fixed somehow. Thanks for all the hard work as always!

### Reproduction Steps

- Run PCSX2 with all default settings
- Play any mission with mountainous terrain such as 3 or 11.
- Observe edges of screen.
- Setting 16:9 in the in-game options makes the glitching more obvious.

### PCSX2 Revision

v1.7.5645

### Operating System

Windows 11

### CPU

i7-12700H

### GPU

RTX 3060

### Logs & Dumps

[AC04 screen edge texture bug.zip](https://github.com/PCSX2/pcsx2/files/14773086/AC04.screen.edge.texture.bug.zip)

**Comment by PositronCannon** [comment]
Unfortunately the fix [is only partial at best](https://www.youtube.com/watch?v=sbArUDgfqiA).

If I'm reading the [commit](https://github.com/PCSX2/pcsx2/pull/10962/commits/f9f88c1ad40249e0a22d27147d118de4aea7e634#diff-0fb2033b392553bd4932fd1ce069528d7a64f878d7feb8bca06ab377a8ce9cb7R57210) right, the relevant changes for SLUS-20152 are enabling full mipmaps and trilinear filtering. I also tried enabling those manually and all sorts of combinations, just in case the automatic settings were not being applied for whatever reason, and while full mipmaps do reduce the amount of glitched tiles (easy to see when the game is paused and toggling between settings), there's still a lot left, especially in first person view.

I've attached a new GS dump from the new version just in case:
[AC04 screen edges texture glitch (1.7.5652 update).zip](https://github.com/PCSX2/pcsx2/files/14792747/AC04.screen.edges.texture.glitch.1.7.5652.update.zip)

**Comment by JordanTheToaster** [comment]
> Unfortunately the fix [is only partial at best](https://www.youtube.com/watch?v=sbArUDgfqiA).
> 
> If I'm reading the [commit](https://github.com/PCSX2/pcsx2/pull/10962/commits/f9f88c1ad40249e0a22d27147d118de4aea7e634#diff-0fb2033b392553bd4932fd1ce069528d7a64f878d7feb8bca06ab377a8ce9cb7R57210) right, the relevant changes for SLUS-20152 are enabling full mipmaps and trilinear filtering. I also tried enabling those manually and all sorts of combinations, just in case the automatic settings were not being applied for whatever reason, and while full mipmaps do reduce the amount of glitched tiles (easy to see when the game is paused and toggling between settings), there's still a lot left, especially in first person view.
> 
> I've attached a new GS dump from the new version just in case: [AC04 screen edges texture glitch (1.7.5652 update).zip](https://github.com/PCSX2/pcsx2/files/14792747/AC04.screen.edges.texture.glitch.1.7.5652.update.zip)

Enable manual fixes keep everything the same as it is in the db currently and enable Texture Inside RT to merge targets. Tell me if that fixes it.

**Comment by PositronCannon** [comment]
Setting Texture Inside RT to Merge Targets does fix the missing (black) textures, however the rest of the glitchy ones still remain.

With setting disabled:

![imagen](https://github.com/PCSX2/pcsx2/assets/65429992/e6437d70-fc5a-4963-9511-7079b3b076d3)

With setting enabled: 

![imagen](https://github.com/PCSX2/pcsx2/assets/65429992/62fa2836-bfbf-464c-8253-72ea2d001b3d)

Video showing the remaining glitches: https://www.youtube.com/watch?v=6FjZVYfEQ1w

On the bright side, the glitching is almost gone in 3rd person now at least. I guess this is the negative side of a rare PS2 game that actually does proper widescreen natively rather than Vert- (like, say, the other 2 AC games). 😅

**Comment by PositronCannon** [comment]
Setting Texture Inside RT to Merge Targets does fix most of the missing (black) textures, however the rest of the glitchy ones still remain.

With setting disabled:

![imagen](https://github.com/PCSX2/pcsx2/assets/65429992/905d5391-4504-4c93-a1d8-8011eb02265f)

With setting enabled:

![imagen](https://github.com/PCSX2/pcsx2/assets/65429992/2978981f-4715-4159-bf1c-e7df9ae52a37)

Video showing the remaining glitches: https://www.youtube.com/watch?v=6FjZVYfEQ1w

On the bright side, the glitching is almost gone in 3rd person now at least. I guess this is the negative side of a rare PS2 game that actually does proper widescreen natively rather than Vert- (like, say, the other 2 AC games). 😅

**Comment by refractionpcsx2** [comment]
Is that with Mipmapping + Trilinear enabled also?

**Comment by PositronCannon** [comment]
> Is that with Mipmapping + Trilinear enabled also?

Yep. I had left them on automatic before, but just in case I just tested manually setting mipmaps to full and trilinear to both PS2 and Forced and it's the same story in all cases. As mentioned, full mipmaps do improve things over Basic, but it's still quite glitchy.

**Comment by refractionpcsx2** [comment]
Can we get a new GS dump when it's glitching?

I suspect this is gonna be a bit of whack-a-mole..

**Comment by PositronCannon** [comment]
Here you go:

[AC04 screen edge textures glitch (full mipmaps+trilinear).zip](https://github.com/PCSX2/pcsx2/files/14794048/AC04.screen.edge.textures.glitch.full.mipmaps%2Btrilinear.zip)

**Comment by EdEddED** [comment]
I've encountered this same issue on the Japanese Version of the game [SLPS-25052]. I've tried different rendering pipelines alongside what was suggested above, and got an interesting result out of it. The issue is a lot more minimized when I'm using the Vulkan renderer, though it is still present if you know where to look. However, on the Auto setting [which in my case, is Metal, being on macOS] The issue is as clear as day on the edges of the screen.

I'll attach screenshots to this post - let me know if you would also like a GS dump from my end. In my case, the issue is on the mountains on the far left of the image.

Vulkan Renderer
![Terrain Issue (Vulkan Backend)](https://github.com/PCSX2/pcsx2/assets/38295703/68eed540-a1c3-434e-a702-e336c906b330)


Metal Renderer
![Terrain issue (Metal Backend)](https://github.com/PCSX2/pcsx2/assets/38295703/4aa47c6c-4168-4f96-b588-ec1ba459dcfe)

My specs: 
PCSX2 Revison v1.7.5670
macOS Sonoma (ver. 14.3.1)
Apple M1 Chip (Does both CPU and GPU work)

**Comment by mirh** [comment]
Is this just a variation of #5347?

**Comment by refractionpcsx2** [comment]
I believe this is a mipmapping issue in hardware picking the wrong level, as mirh said.
